### PR TITLE
Upload and process Plaso storage files from the UI

### PR DESCRIPTION
**Pull request by berggren**
### Upload Plaso file and run psort automatically

This adds the feature to upload a Plaso storage file that will be picked up by Celery and processed.
There are some funky stuff in there, because how Celery and Flask plays together. Let me know if you need more info.

**Note:** This is the first take on this. Tests for the celery stuff will be added in a later PR because it's a pain.

---

Ref issue: #52 
Reviewer: @onager

### CODE REVIEW

File: timesketch.conf [hunk 1]
```diff
@@ -36,3 +36,6 @@ ELASTIC_PORT = 9200
 # another name you can configure that here.
 SSO_ENABLED = False
 SSO_USER_ENV_VARIABLE = u'REMOTE_USER'
+
+# Upload folder for Plaso dump files to be processed.
```
Comment (Onager) [inline timesketch.conf]: Can you expand this a bit? This is where storage files will be parked temporarily before they're inserted into elastic, right?

File: timesketch/__init__.py [hunk 2]
```diff
@@ -121,3 +138,29 @@ def load_user(user_id):
...
+    celery.conf.update(app.config)
+    TaskBase = celery.Task
+
+    # pylint: disable=no-init
+    class ContextTask(TaskBase):
+        """Add Flask context to the Celery tasks created."""
+        abstract = True
+
+        def __call__(self, *args, **kwargs):
+            """Return Task withing a Flask app context.
```
Comment (Onager) [inline timesketch/__init__.py]: "within"

File: timesketch/api/v1/resources.py [hunk 3]
```diff
@@ -479,3 +484,79 @@ def _set_label(label, toggle=False):
             return self.to_json(
                 annotation, status_code=HTTP_STATUS_CODE_CREATED)
         return abort(HTTP_STATUS_CODE_BAD_REQUEST)
+
+
+class UploadFileResource(ResourceMixin, Resource):
+    """Resource thar processes uploaded files."""
```
Comment (Onager) [inline timesketch/api/v1/resources.py]: ?

File: timesketch/lib/tasks.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        options.flush_interval = 1000
+        options.index = index_name
+        options.name = timeline_name
+        options.owner = None
+        options.storage_file = source_file_path
+
+        psort_frontend = PsortCeleryFrontend()
+        psort_frontend.ParseOptions(options)
+        counter = psort_frontend.ProcessStorage(options, None)
```
Comment (Onager) [inline timesketch/lib/tasks.py]: This looks like it's using an old version of plaso/psort - is that correct?

File: timesketch/lib/tasks.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    Args:
+        source_file_path: Path to plaso storage file.
+        timeline_name: Name of the Timesketch timeline.
+        index_name: Name of the datastore index.
+
+    Returns:
+        Dictionary with count of processed events.
+    """
+    l2c = PlasoCeleryTask()
```
Comment (Onager) [inline timesketch/lib/tasks.py]: why l2c?

File: timesketch.conf (hunk 1 above)
Comment (berggren) [inline timesketch.conf]: Correct. I clarified a bit.

File: timesketch/__init__.py (hunk 2 above)
Comment (berggren) [inline timesketch/__init__.py]: Done.

File: timesketch/api/v1/resources.py (hunk 3 above)
Comment (berggren) [inline timesketch/api/v1/resources.py]: Done.

File: timesketch/lib/tasks.py (hunk 5 above)
Comment (berggren) [inline timesketch/lib/tasks.py]: :) "log2celery".. you see.. log2timeline.. but yeah, this is an old remnant of code long gone. I have changed this.

File: timesketch/lib/tasks.py (hunk 4 above)
Comment (berggren) [inline timesketch/lib/tasks.py]: Maybe? Like 7 days old.. but with current Plaso work this is like a 1000 years old.. :) I will pull the latest and test this and adjust. Thanks for the heads up.

Comment (berggren) [inline timesketch/lib/tasks.py]: I wonder what have changed..... ;)

**Comment by Onager** [comment]
LGTM

**Pull request merged**